packages/blocks/src/api/utils.js

```diff
@@ -296,3 +296,27 @@ export function __experimentalGetBlockAttributesNamesByRole( name, role ) {
 			attributes[ attributeName ]?.__experimentalRole === role
 	);
 }
+
+/**
+ * Given block attributes, return only those which do not have the `internal`
+ * role.
+ *
+ * @param {string} name       The block's name
+ * @param {Object} attributes The block's attributes
+ * @return {Object} The attributes, stripped of any internal attributes.
+ */
+export function __experimentalStripInternalBlockAttributes( name, attributes ) {
```

Comment:
I feel this should be something more generic to remove/strip attributes by `role`.

Reply:
Do we have other use cases at the moment? As long as it's an experimental API we don't have to worry about generalization. We can consolidate later.

Reply:
No we don't, so for sure leaving it as is, is fine for now 😄 . I just think that this function will change to handle the case for `copy`. We'll see from the new iterations..